src/lib.rs

```diff
@@ -781,7 +780,7 @@ enum Decoder {
 }
 
 impl Expr {
-    fn eval<'a>(&'a self, scope: &'a mut Scope) -> Cow<'a, Value> {
+    fn eval<'a>(&'a self, scope: &'a Scope<'a>) -> Cow<'a, Value> {
```

Comment:
If we have enough `&'a Scope<'a>`, it might be worth considering a helper alias `type ScopeRef<'a> = &'a Scope<'a>`.